src/api/spotify/auth.ts

```diff
@@ -46,3 +49,40 @@ export function login() {
 	});
 	window.location.replace(`https://accounts.spotify.com/authorize?${oAuthParams}`);
 }
+
+export function exchangeToken(code: string): void {
+	const verifier = localStorage.getItem('code_verifier');
+	const body = new URLSearchParams({
+		code: code,
+		redirect_uri: redirectUri as string,
+		grant_type: 'authorization_code',
+		client_id: clientId as string,
+		code_verifier: verifier as string
+	});
+	const xhr = new XMLHttpRequest();
+	xhr.open('POST', 'https://accounts.spotify.com/api/token', true);
+	xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+	xhr.setRequestHeader('Authorization', 'Basic ' + btoa(clientId + ':' + clientSecret));
+	xhr.send(body);
+	xhr.onload = handleExchangeTokenResponse;
```

Comment:
I thought we were using fetch, not xml?

Reply:
Didn’t want too many changes on one or so was going to make one dedicated to change from using xhr to fetch